connect_4/lib/connect_4_web/channels/room_channel.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Connect4Web.RoomChannel do
+
+  use Phoenix.Channel
+
+  def join("c4:common", _message, socket) do
+    game = Game.new_game()
+    socket =
+      socket
+      |> assign(:game, game)
+    { :ok, socket }
+  end
+
+  def handle_in("turn_played", msg, socket) do
+    game = socket.assigns[:game];
+    g = Game.play(game, msg);
+    socket =
+      socket
+      |> assign(:game, g)
+    broadcast!(socket, "echo", g)
```

Comment:
The use of broadcast here is actually a bug, as it broadcasts the game state to all connected browsers. (Try running the game in two browser windows and making a move in one of them)

# -4